test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua

```diff
@@ -0,0 +1,75 @@
+local net = require('net.box')
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+    g.server:exec(function()
+        box.schema.create_space('test', {
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"},
+            },
+        })
+
+        box.space.test:create_index('primary', {
+            type = 'tree',
+            unique = true,
+            parts = {1, 'unsigned'},
+        })
+
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = true,
+            parts = {2, 'string'},
+        })
+    end)
+end)
+
+g.after_all(function()
+    g.server:drop()
+end)
+
+g.before_each(function()
+    -- fetch schema = false to resolve index on server instead of client
+    g.conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(g.conn, nil)
```

Comment:
This check is redundant. Your connection can never be `nil`. If an error occurs during the connection construction, an exception is thrown. Since you do not specify the `wait_connected` option (which is totally fine for a test), you are by default blocked until the connection is established [^1], so you do not have to care about this:
> wait_connected: a connection timeout. By default, the connection is blocked until the connection is established, but if you specify wait_connected=false, the connection returns immediately. If you specify this timeout, it will wait before returning (wait_connected=1.5 makes it wait at most 1.5 seconds).

[^1]: https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/#lua-function.net_box.connect